Sources/Confidence/Confidence.swift

```diff
@@ -26,20 +43,25 @@ public class Confidence: ConfidenceEventSender {
     }
 
     public func updateContextEntry(key: String, value: ConfidenceValue) {
-        context[key] = value
+        localContext[key] = value
     }
 
     public func removeContextEntry(key: String) {
-        context.removeValue(forKey: key)
+        localContext.removeValue(forKey: key)
```

Comment:
here we should also add the key to a set of strings (which holds the removed keys) to filter it later in context computation.